packages/flutter/lib/src/rendering/object.dart

```diff
@@ -167,6 +170,22 @@ class PaintingContext extends ClipContext {
     }());
   }
 
+  /// Used for tests that directly invoke [RenderObject.paint] or
+  /// [RenderObject.debugPaint].
```

Comment:
this documentation should still say what it does (including what it does in release builds)

Reply:
Done. Also updated docs on the canvas getter to explain what tests might need to do if they directly invoke paint/debugPaint.